Add XML stylesheet example with XSLT

jhabdas opened the pull request:
Just an example for #118.

benbalter commented on lib/feed.xml:
```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
+<?xml-stylesheet type="text/xml" href="feed.xslt.xml"?>
```
This should be `{{ url_base}}/feed.xslt.xml`

jhabdas commented on lib/feed.xml:
```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
+<?xml-stylesheet type="text/xml" href="{{ url_base }}/feed.xslt.xml"?>
```
Updated to use `url_base` liquid object.

jhabdas commented:
Working on some basic customizations to make the default XSLT template look a little nicer and be more usable.

jhabdas commented:
Made a small adjustment to the XSLT stylesheet so it outputs both the published and updated dates, and fixed a unit test for `url_base` which I may have missed an edge case for.

Working with the XSLT is tricky so I opted not to go too much further. Ideally some basic date parsing via use of [`format-dateTime`](https://www.w3.org/TR/xpath-functions-30/#func-format-dateTime) could be helpful for displaying dates. If someone wanted to go all out they could adapt [the code](https://plugins.svn.wordpress.org/xslt/trunk/) for [Better RSS Feeds](https://wordpress.org/plugins/xslt/screenshots/) to output a fully styled HTML document with hyperlinks and social sharing. For my purposes I'm happy with simply not seeing unformatted XML.

benbalter commented:
> Made a small adjustment to the XSLT stylesheet so it outputs both the published and updated dates, and fixed a unit test for url_base which I may have missed an edge case for.

Would you mind posting a screenshot?

Also /cc @jglovier famed Jekyll designer, in case he has any suggestions.

benbalter commented on lib/jekyll/jekyll-feed.rb:
```diff
@@ -47,5 +50,30 @@ def feed_exists?
+    end
+
+    # Checks if a feed stylesheet exists in the site source
+    def xslt_exists?
```
Should this an the `feed_exists?` methods be abstracted into a `file_exists?(file)` method could be shared for both?

benbalter commented on lib/jekyll/jekyll-feed.rb:
```diff
@@ -47,5 +50,30 @@ def feed_exists?
+      File.expand_path "../feed.xslt.xml", File.dirname(__FILE__)
+    end
+
+    def xslt_content
```
Same with `content_for_file(file)` or similar. Just trying to DRY things up where we can.

benbalter commented on spec/jekyll-feed_spec.rb:
```diff
@@ -249,6 +249,12 @@
     end
   end
 
+  context "feed stylesheet" do
```
We'll also want to test that the resulting site includes the stylesheet file.

jhabdas replied:
Could you provide a little more direction on this one? I believe I was able to tackle the other two refactorings.

jhabdas replied:
Disregard this request. I think I got it.

jhabdas commented:
Here's a screenshot for the basic example provided. It made it immediately clear I wasn't outputting my RSS updated/modified dates properly from my Jekyll site:

![screen shot 2016-06-13 at 2 46 04 pm](https://cloud.githubusercontent.com/assets/440298/16020981/f3970392-3175-11e6-84f0-052864834e8f.png).

I'll take a stab at those refactorings.

jhabdas commented:
@benbalter Ready for another review pass.

benbalter commented:
@pathawks @parkr any strong feelings here?

pathawks commented:
LGTM :shipit:

parkr commented:
Fine with me!

benbalter commented:
@jekyllbot: merge

benbalter commented:
@jhabdas Thanks for this!

jhabdas commented:
My pleasure. This was my second time writing teh Ruby. Thanks for making it fun @benbalter. This is almost better than humans.txt!
